packages/server/tests/views.test.ts

```diff
@@ -0,0 +1,266 @@
+import { AdxUri } from '@adxp/common'
+import { MicroblogClient, Post } from '@adxp/microblog'
+import { users, posts, replies } from './test-data'
+import getPort from 'get-port'
+import * as util from './_util'
+
+const USE_TEST_SERVER = true
+
+let alicePosts: AdxUri[] = []
+let bobPosts: AdxUri[] = []
+let carolPosts: AdxUri[] = []
+let danPosts: AdxUri[] = []
+let bobFollows: Record<string, AdxUri> = {}
+let bobLikes: Record<string, AdxUri> = {}
+let badges: AdxUri[] = []
+
+describe('pds views', () => {
+  let alice, bob, carol, dan: MicroblogClient
+  let closeFn: util.CloseFn
+
+  beforeAll(async () => {
+    const port = USE_TEST_SERVER ? await getPort() : 2583
+    closeFn = await util.runTestServer(port)
+
+    const url = `http://localhost:${port}`
+    alice = new MicroblogClient(url, users.alice.did)
+    bob = new MicroblogClient(url, users.bob.did)
+    carol = new MicroblogClient(url, users.carol.did)
+    dan = new MicroblogClient(url, users.dan.did)
+  })
+
+  afterAll(async () => {
+    await closeFn()
+  })
+
+  it('register users', async () => {
+    await alice.register(users.alice.name)
+    await bob.register(users.bob.name)
+    await carol.register(users.carol.name)
+    await dan.register(users.dan.name)
+  })
+
+  it('creates profiles', async () => {
+    await alice.createProfile(users.alice.displayName, users.alice.description)
+    await bob.createProfile(users.bob.displayName, users.bob.description)
+  })
+
+  it('follow people', async () => {
+    await alice.followUser(bob.did)
+    await alice.followUser(carol.did)
+    await alice.followUser(dan.did)
+    bobFollows[alice.did] = await bob.followUser(alice.did)
+    bobFollows[carol.did] = await bob.followUser(carol.did)
+    await carol.followUser(alice.did)
+    await dan.followUser(bob.did)
+  })
+
+  it('makes some posts', async () => {
+    const alice0 = await alice.createPost(posts.alice[0])
+    const bob0 = await bob.createPost(posts.bob[0])
+    const carol0 = await carol.createPost(posts.carol[0])
+    const dan0 = await dan.createPost(posts.dan[0])
+    const dan1 = await dan.createPost(posts.dan[1], [
+      {
```

Comment:
not in love with this way of creating user mentions, but also don't see a great alternative. thoughts?